### PR TITLE
apply `black` formatting to all of dmoj branch

### PR DESCRIPTION
**Pull request by Riolku**
This applies formatting changes by `black` to the entire branch. We should probably get formatting checks up with `flake8` for some of these things.

### CODE REVIEW

**Comment by Xyene** [comment]
We use black 19.03 (see the workflow yml), for the avoidance of doubt do we still have diffs with that version?

**Comment by Riolku** [comment]
Re-installed the exact version and re-ran black

**Review by quantum5 — APPROVED** [review]
Changes look reasonable.

**Comment by Riolku** [comment]
https://pypi.org/project/flake8-black/ going to throw this in this PR later

**Comment by quantum5** [comment]
ok to test

**Comment by Riolku** [comment]
Added a flake8-black package to run checks. Also ran `black` against the testsuite, which definitely isn't necessary, and flake8 won't lint there, but the changes are probably still fine. Sidenote: why don't we lint the testsuite?

File: testsuite/bridged_checker/tests/py_ac/ac.py [hunk 1]
```diff
@@ -1 +1 @@
-print -1, 1
+print - 1, 1
```
Comment (quantum5) [inline testsuite/bridged_checker/tests/py_ac/ac.py:1]: Should we be running black on the testsuite? This is it attempting to apply Python 3 formatting on Python 2 code.

Comment (Riolku) [reply testsuite/bridged_checker/tests/py_ac/ac.py:1]: Yikes... can we configure black somehow? Right now it takes a directory. We can apply to dmoj, but then there are some other files that deserve to be scanned... I'll do some research

Comment (Riolku) [reply testsuite/bridged_checker/tests/py_ac/ac.py:1]: We have options to install `black[python2]`, but I think that's not the solution here.

Comment (quantum5) [reply testsuite/bridged_checker/tests/py_ac/ac.py:1]: Definitely shouldn't be formatting the test suite. We could have deliberate syntax errors and other stuff. Should probably just exclude the entire directory from black and flake8.

Comment (Riolku) [reply testsuite/bridged_checker/tests/py_ac/ac.py:1]: flake8 already excludes it. We can probably just run `black` with `--exclude testsuite`. I'll try that

Comment (Riolku) [reply testsuite/bridged_checker/tests/py_ac/ac.py]: The command is `--exclude testsuite/`, since it seems to be a regex, it will ignore `testsuite.py`

Comment (Riolku) [reply testsuite/bridged_checker/tests/py_ac/ac.py]: But happily, it will be caught by `flake8`.

**Review by quantum5 — COMMENTED** [review]
Seems ok now. Maybe try putting an exclusion in `pyproject.toml` to avoid having to exclude the testsuite every time?

**Comment by Riolku** [comment]
Should we be skipping string normalization with black? Seems we could benefit from it, if #888 is any indication

**Comment by quantum5** [comment]
If it's fine post-#888 then we can remove the skip.

File: pyproject.toml [hunk 2]
```diff
@@ -3,4 +3,5 @@ requires = ["setuptools", "wheel", "Cython"]
 
 [tool.black]
 line-length = 120
-skip-string-normalization = true
+skip-string-normalization = false
+exclude = '^/testsuite'
```
Comment (quantum5) [inline pyproject.toml:7]: Should probably be `extend-exclude` to avoid removing the default exclusions.

**Comment by Riolku** [comment]
It is very not fine. It seems to prefer double quotes? What should we do? We can definitely let black run and do its thing, with the benefit of having it be able to auto-format for us, or we can stick to our current styling.

File: pyproject.toml (hunk 2 above)
Comment (Riolku) [reply pyproject.toml:7]: That did not work at all for me locally. Not sure why. Is black outdated for us?

**Comment by quantum5** [comment]
I'd rather have single quotes and enforce them with `flake8-quotes`.

File: pyproject.toml (hunk 2 above)
Comment (Riolku) [reply pyproject.toml:7]: Works fine under 21.8b0 on Debian 11, but not 19 or whatever the version we have in the workflow is. What do we want to do?

Comment (quantum5) [reply pyproject.toml:7]: Yes. Try upgrading to `21.8b0` and see if it breaks anything. If not, let's just upgrade.

**Review by quantum5 — APPROVED** [review]

**Pull request merged**